core/src/value/deserialize.rs

```diff
@@ -29,55 +50,286 @@ pub enum Error {
     InvalidNumberToken { value: u8, backtrace: Backtrace },
     #[snafu(display("Invalid time zone sign token: got '{}', but must be '+' or '-'", *value as char))]
     InvalidTimeZoneSignToken { value: u8, backtrace: Backtrace },
+    #[snafu(display(
+        "Could not parse incomplete value: first missing component: {:?}",
+        component
+    ))]
+    IncompleteValue {
+        component: DateComponent,
+        backtrace: Backtrace,
+    },
+    #[snafu(display("Range is zero"))]
+    ZeroRange { backtrace: Backtrace },
 }
 
 type Result<T, E = Error> = std::result::Result<T, E>;
 
+#[derive(Debug, PartialEq)]
+pub enum DateComponent {
+    Year,
+    Month,
+    Day,
+    Hour,
+    Minute,
+    Second,
+    Fraction,
+    UTCOffset,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum DatePrecision {
+    Year(u16),
+    Month(u16, u8),
+    Day(u16, u8, u8),
+}
+
+#[derive(Debug, PartialEq)]
+pub struct PartialDate {
```

Comment:
The creation of both `PartialDate` and a `DatePrecision` is a bit intriguing. Couldn't `DatePrecision` be renamed to `PartialDate` and fulfill this one's role all the same?